File: tut01.py
import numpy as np
import matplotlib.pyplot as plt

PI = np.pi
################################################
################# Question 11 ##################
################################################
s,T = -0.1 + 2*PI*np.complex(0,1),1
L = np.arange(-6,6)
y = np.exp(L*s*T)
y = np.real(y)
plt.plot(L,y)
plt.show()

################################################
################# Question 12 ##################
################################################
f0 = 50
T = 1000*10**(-6)
L = np.linspace(0,40*10**(-3),int(40*10**(-3)/T))
y = np.cos(2*PI*f0*L)
markerline, stemlines, baseline = plt.stem(L, y, '-')
plt.setp(markerline, 'markerfacecolor', 'b')
plt.setp(baseline, 'color','r', 'linewidth', 2)
plt.plot(L,y)
plt.show()

################################################
################# Question 13 ##################
################################################
L = np.arange(-10,11,1)
y1 = (PI/6.0)*L-(PI/3.0)
y1 = ((0.9)**L)*np.cos(y1)
y2 = (49*PI/6.0)*L-(PI/3.0)
y2 = ((1.1)**L)*np.cos(y2)
plt.plot(L,y1)
plt.plot(L,y2)
plt.show()

################################################
################# Question 14 ##################
################################################
L = np.arange(0,1,0.001)
w0 = 20*PI
PHI = 0
a,A = 6,60
y = A*np.sin((w0)*L+PHI)*np.exp(-a*L)
plt.plot(L,y)
plt.show()

################################################
################# Question 15 ##################
################################################

L = np.arange(0,4.01,0.01)
y1 = np.sin(0.25*L*2*PI)
y2 = np.sin(1.25*L*2*PI)
plt.plot(L,y1)
plt.plot(L,y2)
plt.show()

L = np.arange(0,51,1)
y = np.cos(2*PI*PI*L)
plt.plot(L,y)
plt.show()

################################################
################# Question 16 ##################
################################################
f = 0.5
L = np.arange(0,10.01,0.01)
y0 = np.sin(2*PI*f*L)
y1 = np.sin(2*PI*f*L*0.5)
y2 = np.sin(2*PI*f*L*2)
y3 = np.sin(-2*PI*f*L)
y4 = np.sin(2*PI*f*(L - 5))
plt.subplot(2,3,1)
plt.plot(L,y0)
plt.subplot(2,3,2)
plt.plot(L,y1)
plt.subplot(2,3,3)
plt.plot(L,y2)
plt.subplot(2,3,4)
plt.plot(L,y3)
plt.subplot(2,3,5)
plt.plot(L,y4)
plt.show()

File: tut01.m
% X = [-5:1:5];
% T = 1;
% s = -0.1 + 2*pi*j;
% y = exp(s*T.*X);
% y = real(y);
% plot(X,y)

% T = 40*10^(-3);
% t = 10000*10^(-6);
% X = [0:t:T];
% f = 50;
% y = cos(2*pi.*X*f);
% stem(X,y)

% X = [-10:1:10];
% y_1 = 0.9.^X;
% y_2 = cos((pi/6.0)*X - (pi/3.0));
% y1 = y_1.*y_2;
% plot(X,y1)

% X = [-10:1:10];
% y_1 = 1.1.^X;
% y_2 = cos((pi/6.0)*49*X - (pi/3.0));
% y1 = y_1.*y_2;
% plot(X,y1)

% t = [0:0.001:1];
% phi = 0;
% a = 6;
% A = 60;
% w0 = 20*pi;
% y = A*sin(w0*t + phi).*(exp(-a*t));
% plot(t,y)

% X = [0:0.01:4];
% y1 = sin(2*pi*0.25*X);
% y2 = sin(2*pi*1.25*X);
% plot(X,y1);
% hold on;
% plot(X,y2)

% X = [0:1:50];
% y = cos(2*pi*pi*X);
% plot(X,y)

X = [0:0.01:2];
f = 1;
y1 = sin(2*pi*f*X);
y2 = sin(2*2*pi*f*X);
y3 = sin(2*0.5*pi*f*X);
y4 = -sin(2*2*pi*f*X);
y5 = sin(2*2*pi*f*(X-1));
figure
subplot(2,3,1)
plot(X,y1);
subplot(2,3,2)
plot(X,y2);
subplot(2,3,3)
plot(X,y3);
subplot(2,3,4)
plot(X,y4);
subplot(2,3,5)
plot(X,y5);
